### DISCUSSION TITLE
Which code can we refer to if  phase field simulation of selective laser melting

### DISCUSSION THREAD

**Discussion by mikejwg** [idaholab/moose#16554] [Q&A Modules: Phase field] [answered]
Dear everyone,

Anyone can share me the example similar to phase field simulation of selective laser melting, in which cooling rate and temperature gradient should be considered.
I found  the examples in the phase field directory can not consider  cooling rate and temperature gradient.

thanks.
WG

**Comment by laagesen** [comment, marked as answer]
Hi, we currently don't have an example in the repository for directional solidification. We are doing some work in this area and are hoping to add an input file as an example in the next few months.

**Reply by mikejwg** [reply]
Thanks for your reply.  Look forward to the work.  Hope the input file can consider rapid solidification because the cooling rate is  greater than 1e5 K/s and the temperature gradient is greater than  1e5 K/m during SLM process.

**Reply by SudiptaBiswas** [reply]
@mikejwg The following could be a starting point for what you need, 
https://github.com/idaholab/moose/blob/next/modules/phase_field/examples/anisotropic_interfaces/GrandPotentialPlanarGrowth.i

**Reply by mikejwg** [reply]
Thanks. Could you show me which keywords respect to temperature gradient and cooling rate we should revise in this example for our purpose?

**Reply by SudiptaBiswas** [reply]
You will have to modify the following line for the temperature function, https://github.com/idaholab/moose/blob/next/modules/phase_field/examples/anisotropic_interfaces/GrandPotentialPlanarGrowth.i#L46

**Reply by rl3fz** [reply]
Hello, I hope it's okay if I ask a follow-up here:

@SudiptaBiswas, could you please tell me why you recommend GrandPotentialPlanarGrowth.i here, as opposed to GrandPotentialSolidification.i?

I am trying to simulate solidification from powder bed fusion, a type of selective laser melting, and saw that you recommended these examples in a related issue (#17689), but I chose to use the Solidification model file based only on its name.

**Reply by SudiptaBiswas** [reply]
GrandPotentialPlanarGrowth.i is for directional solidification whereas GrandPotentialSolidification.i is for isothermal solidification. Both use the same kernels and materials, only the initial condition and thermal field are different.

**Reply by rl3fz** [reply]
That makes sense, thank you!

**Comment by ngrilli** [comment]
@mikejwg 
I am not working on phase field, but just simulating the selecting laser melting
coupled with crystal plasticity model.

You can find some code and examples here:
https://github.com/ngrilli/c_pfor_am

Best Regards,
Nicolò Grilli
National University of Singapore

**Reply by mikejwg** [reply]
Thanks very much for your kindness.  Could you share me one completed SLM example (input file)?  Anyway, thanks  a  lot for the codes. 

Best regards,
jiangwugui@gmail.com
